src/server/lib/queryParams.ts

```diff
@@ -45,18 +45,8 @@ export const parseExpressQueryParams = (
     ref,
     encryptedEmail,
     error,
-  }: {
-    returnUrl?: string;
-    clientId?: string;
-    emailVerified?: string;
-    emailSentSuccess?: string;
-    csrfError?: string;
-    recaptchaError?: string;
-    refViewId?: string;
-    ref?: string;
-    encryptedEmail?: string;
-    error?: string;
-  },
+  }: Record<keyof QueryParams, string | undefined>, // parameters from req.query
+  bodyParams: Pick<QueryParams, 'ref' | 'refViewId'> = {}, // some parameters may be manually passed in req.body too
```

Comment:
wondering if it might be nice to extract `'ref' | 'refViewId'` into a type called `AllowedBodyParams` or something along those lines

Reply:
The main reason I didn't extract it into a separate type is that this is the only place that it needs to be used. If it was being used in multiple places then that would be an option.

Reply:
Actually I did end up refactoring it into a shareable `TrackingQueryParams` interface, as we could use the same interface as a type in your previous comment..

Reply:
nice! :)